package in.co.technoflask.projects.desiwanderer.user.service;

import in.co.technoflask.projects.desiwanderer.security.service.SecurityService;
import in.co.technoflask.projects.desiwanderer.user.entity.User;
import in.co.technoflask.projects.desiwanderer.user.exception.UserNotFoundException;
import in.co.technoflask.projects.desiwanderer.user.repository.UserRepository;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@Slf4j
@RequiredArgsConstructor
public class UserQueryService {

  private final UserRepository userRepository;
  private final SecurityService securityService;

  public User findByAuthentication() {
    UUID id = this.securityService.getUserId();
    return this.findById(id);
  }

  public User findById(UUID id) {
    return this.userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(id));
  }
}
